test/transpiler/transpiler_double_test.ts

```diff
@@ -16,31 +16,814 @@ describe('Transpiler', () => {
+                .setSignature('div', WebAssemblyType.FLOAT_64, WebAssemblyType.FLOAT_64, WebAssemblyType.FLOAT_64)
+                .transpile('function div(a, b) { return a / b }');
+            wrapper.setFunctionName('div');
+
+            expect(wrapper.call(1.67, 0.58)).to.closeTo(2.879310344827586, 0.000000000000001);
+            expect(wrapper.call(-5.15, 2.1)).to.closeTo(-2.452380952380952, 0.000000000000001);
+        });
+
+        it('should handle double-int combination division', () => {
+            const content = 'function div(a, b) { return a / b; }';
+            const wrapper = transpiler
+                .setSignature('div', WebAssemblyType.FLOAT_64, WebAssemblyType.INT_32, WebAssemblyType.FLOAT_64)
+                .transpile(content);
+            wrapper.setFunctionName('div');
+
+            expect(wrapper.call(5, 14.9)).to.closeTo(0.3355704697986577, 0.000000000000001);
+            expect(wrapper.call(-100, 12.59)).to.closeTo(-7.942811755361398, 0.000000000000001);
+            expect(wrapper.call(20, -13.01)).to.closeTo(-1.53727901614143, 0.000000000000001);
+            expect(wrapper.call(-9, -2.4)).to.closeTo(3.75, 0.0001);
+
+            const wrapper2 = new Transpiler()
+                .setSignature('div', WebAssemblyType.FLOAT_64, WebAssemblyType.FLOAT_64, WebAssemblyType.INT_32)
+                .transpile(content);
+            wrapper2.setFunctionName('div');
+
+            expect(wrapper2.call(5.31, 4)).to.closeTo(1.3275, 0.000001);
+            expect(wrapper2.call(14.1, -3)).to.closeTo(-4.7, 0.001);
+            expect(wrapper2.call(-23.4, 35)).to.closeTo(-0.6685714285714286, 0.000000000000001);
+            expect(wrapper2.call(-9.5, -2)).to.closeTo(4.75, 0.000000000000001);
+        });
+
+        it('should handle double division with array', () => {
+            const wrapper = transpiler
+                .setSignature('div', WebAssemblyType.FLOAT_64, WebAssemblyType.FLOAT_64, WebAssemblyType.INT_32_ARRAY)
+                .transpile('function div(a, b) { return a / b[0]; }');
+            wrapper.setFunctionName('div');
+
+            expect(wrapper.call(1.13, [1, 2])).to.closeTo(1.13, 0.001);
+
+            const wrapper2 = transpiler
+                .setSignature('div', WebAssemblyType.FLOAT_64, WebAssemblyType.INT_32_ARRAY, WebAssemblyType.FLOAT_64)
+                .transpile('function div(a, b) { return a[0] / b; }');
+            wrapper2.setFunctionName('div');
+
+            expect(wrapper2.call([1, 2], 2.4)).to.closeTo(0.4166666666666667, 0.000000000000001);
+        });
+
+        it('should handle double division by 0', () => {
+            const wrapper = transpiler
+                .setSignature('div', WebAssemblyType.FLOAT_64, WebAssemblyType.FLOAT_64, WebAssemblyType.FLOAT_64)
+                .transpile('function div(a, b) { return a / b }');
+            wrapper.setFunctionName('div');
+
+            expect(wrapper.call(2.52, 0.0)).to.equal(Infinity);
```

Comment:
Integer division by zero throws an exception. Why is this not the case here?

Reply:
I just checked it with nodejs. This is the default behavior of JavaScript.